packages/proof-service/src/domain/create-proof.ts

```diff
@@ -1,31 +1,27 @@
 import { Field, MerkleWitness, ZkProgram } from 'o1js';
 import {
-  DatabaseEngine,
   ModelDbSetting,
   ModelMerkleTree,
   ModelProof,
   ModelQueueTask,
 } from '@zkdb/storage';
 import CircuitFactory from '../circuit/circuit-factory.js';
 import { ObjectId } from 'mongodb';
-import config from '../helper/config.js';
+import logger from '../helper/logger.js';
 
-export async function createProof() {
-  const taskId = process.argv.slice(2)[0];
-
-  const dbEngine = DatabaseEngine.getInstance(config.mongodbUrl);
-  if (!dbEngine.isConnected()) {
-    await dbEngine.connect();
+export async function createProof(taskId: string) {
+  if (taskId === undefined) {
```

Comment:
Strict condition should be applied on important task, otherwise just throw error or exit.

Reply:
done